packages/cli/src/build/bundle.rs

```diff
@@ -377,57 +380,89 @@ impl AppBundle {
 
         let asset_dir = self.build.asset_dir();
 
-        // First, clear the asset dir
-        // todo(jon): cache the asset dir, removing old files and only copying new ones that changed since the last build
-        _ = std::fs::remove_dir_all(&asset_dir);
+        // First, clear the asset dir of any files that don't exist in the new manifest
         _ = create_dir_all(&asset_dir);
+        // Create a set of all the paths that new files will be bundled to
```

Comment:
it might be a good idea to do all this under `tokio::spawn_blocking()` now that we might be doing some serious optimizations.

Reply:
Good catch, I didn't notice that got removed when assets were removed.